### DISCUSSION TITLE
external MQTT broker, teslamate does not send MQTT_USERNAME

### DISCUSSION THREAD

**Discussion by HenkV74** [teslamate-org/teslamate#3823] [Q&A]
Hi, i enabled in the docker-compose.yml 

      - MQTT_HOST=192.168.1.200
      - MQTT_PORT=1883
      - MQTT_TLS=true
      - MQTT_TLS_ACCEPT_INVALID_CERTS=true
      - MQTT_USERNAME=teslamate
      - MQTT_PASSWORD=pass

However if I look in the mosquitto log it says:
      - New client connected from 172.18.0.2:55520 as TESLAMATE_302E3731333 (p2, c1, k60).
It lacks the u'teslamate'. In this case the updated messages are denied. 

If I connect with the tesla credentials via a mqtt explorer, the mosquitto log says:
     - New client connected from 192.168.1.191:55995 as mqtt-explorer-08aaaefb (p2, c1, k60, u'teslamate').
This is the way and messages are accepted.

Is this a teslamate bug or must I put somewhere else (in grafana?) the mqtt userid and password?

Regards,
Henk

**Comment by HenkV74** [comment]
... At one point in time things started to behave as projected and the problem dissolved. The biggest change I made was  disabling another mqtt service (snap version) which was unknowingly running. Not sure if this was the reason. It had been auto-installed for one or another purpose. So consider this question solved!

**Reply by git-who-hub** [reply]
Hi, how did to sort the issue. I’m facing same issue. Teslamate data is not availed in home assistant. It did get the data when I installed, but after that no new data is being updated

**Reply by kanokc** [reply]
> ... At one point in time things started to behave as projected and the problem dissolved. The biggest change I made was disabling another mqtt service (snap version) which was unknowingly running. Not sure if this was the reason. It had been auto-installed for one or another purpose. So consider this question solved!

Hi,

I just installed Teslamate yesterday and found similar problem.  Using the same credential works on mqtt-explorer but not on TeslaMate.  How did you disable another mqtt service you mentioned?  

Thanks,
Kanok

**Comment by brianmay** [comment]
Sorry, no idea why it isn't working for you. All I can say is it works fine for me...

**Comment by nic0dk** [comment]
you need to add the environment parameters under the correct docker instant, no the mosquitto but instead teslamate instant :)

**Comment by MarcFinns** [comment]
was this ever clarified / solved? i have been banging my heat into this corner for quite a while and no joy..
